api/types/kustomization.go

```diff
@@ -190,6 +193,11 @@ func (k *Kustomization) FixKustomizationPreMarshalling() {
 	// PatchesJson6902 should be under the Patches field.
 	k.Patches = append(k.Patches, k.PatchesJson6902...)
 	k.PatchesJson6902 = nil
+
+	if l := labelFromCommonLabels(k.CommonLabels); l != nil {
+		k.Labels = append(k.Labels, *l)
+		k.CommonLabels = nil
+	}
```

Comment:
Consider returning an error if labels collide.

Also, should probably add a note as to why this isn't done in PostUnmarshalling (because the `edit` commands only work with `commonLabels`, not labels`).

Reply:
Added the check